packages/components/src/components/AssetLogo/AssetLogo.tsx

```diff
@@ -0,0 +1,133 @@
+import styled from 'styled-components';
+import { FrameProps, FramePropsKeys, withFrameProps } from '../../utils/frameProps';
+import { SkeletonCircle } from '../skeletons/SkeletonCircle';
+import { useCallback, useState } from 'react';
+import { borders } from '@trezor/theme';
+import { AssetInitials } from './AssetInitials';
+import { useEffect } from 'react';
+import { makePropsTransient, TransientProps } from '../../utils/transientProps';
+
+const ICONS_URL_BASE = 'https://data.trezor.io/suite/icons/coins/';
+
+export const allowedAssetLogoSizes = [20, 24];
+type AssetLogoSize = (typeof allowedAssetLogoSizes)[number];
+
+export const allowedAssetLogoFrameProps: FramePropsKeys[] = ['margin'];
+type AllowedFrameProps = Pick<FrameProps, (typeof allowedAssetLogoFrameProps)[number]>;
+
+export type AssetLogoProps = AllowedFrameProps & {
+    size: AssetLogoSize;
+    coingeckoId: string;
+    contractAddress?: string;
+    shouldTryToFetch?: boolean;
+    placeholder: string;
+    'data-testid'?: string;
+};
+
+const Container = styled.div<TransientProps<AllowedFrameProps> & { $size: number }>(
+    ({ $size }) => `
+        width: ${$size}px;
+        height: ${$size}px;
+
+        ${withFrameProps}
+    `,
+);
+
+const Logo = styled.img<{ $size: number; $isVisible: boolean }>(
+    ({ $size, $isVisible }) => `
+        width: ${$size}px;
+        height: ${$size}px;
+        border-radius: ${borders.radii.full};
+        visibility: ${$isVisible ? 'visible' : 'hidden'};
+    `,
+);
+
+export const AssetLogo = ({
+    size,
+    coingeckoId,
+    contractAddress,
+    shouldTryToFetch = true,
+    placeholder,
+    margin,
+    'data-testid': dataTest,
+}: AssetLogoProps) => {
+    const [isLoading, setIsLoading] = useState(true);
+    const [isLogoRendered, setIsLogoRendered] = useState<boolean | null>(null);
+
+    const fileName = contractAddress ? `${coingeckoId}--${contractAddress}` : coingeckoId;
+    const getAssetLogoUrl = useCallback(
```

Comment:
Can't this be defined outside the component without the callback and with a simple param?